tests/Unittest-README.md

```diff
@@ -0,0 +1,159 @@
+- `apply_mock()`: Factory function to return a method to apply mocker.patch() on paths
+- `mock_cvpClient()`: mock cvprac classes and objects
+
+For more info on pytest fixtures refer to [this](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html) link
+
+#### How to use Fixtures
+A test function can request a fixture by declaring them as arguments in the function definition. For eg:
+```python
+def test_foo(apply_mock):
+  mock_foo = apply_mock("path.to.original.foo")
+  ...
+```
+
+### Setup and Teardown functions
+Setup and Teardown functions are used to set up and clear up state for testing. They are executed before and after every test in a test suite. 
+
+#### How to write setup and teardown function
+Setup/Teardown functions can be written as a fixture and called at the beginning and end of each test.
+```python
+@pytest.fixture
+def setup(fixture1, fixture2):
+  # use fixture1 and fixture2
+  ...
+  m1 = fixture1(...)
+  m2 = fixture2(...)
+  ...
+  return m1, m2
+
+def test_foo(setup, ...):
+  mock1, mock2 = setup
+  ...
+```
+
+### Parameterizing the tests
+Pytest allows to easily parameterize tests in which case they will be called multiple times, each time with different set of inputs. 
+
+Use the decorator `@pytest.mark.parameterize` to parametrize arguments for a test function.
+
+```python
+@pytest.mark.parametrize("arg1, arg2",
+        [pytest.param(
+            input1, input2,
+            id="scenario1"
+        ),
+        pytest.param(
+            input3, input4,
+            id="scenario2"
+        ),
+        pytest.param(
+            input5, input6,
+            id="scenario3"
+        ),
+        ])
+    def test_foo(self, fixture1, arg1, arg2):
+      ...
+```
+
+## Run tests
+
+### With initial Pytest CLI
+
+```bash
+# Configlet Unit testing
+pytest -rA --cov-report term:skip-covered -v --cov-report term:skip-covered \
+ --html=report.html --self-contained-html --cov-report=html --color yes \
+ --cov=ansible_collections.arista.cvp.plugins.module_utils -m 'generic or api'\
+ unit/<test-file-name>.py
+```
+
+### Makefile usage
+
+```bash
+# Configlet Unit testing
+$ make test TESTS=unit/test_configlet_input.py
+
+# Run all tests related to configlets with logging in CLI set to INFO
+make test TAG='configlet' CLI_LOGGING=INFO
+
+# Run all tests related to configlets with logging in CLI set to INFO and pytest.log set to WARNING
+make test TAG='configlet' CLI_LOGGING=INFO PYTEST_LOGGING=WARNING
+```
+
+### Custom Makefile options
+
+- `TESTS`: Which tests to run. (By default `.`)
+- `TAG`: Which tag to run. (By default `generic or api`)
+- `CLI_LOGGING`: Log verbosity print out to your console
+- `PYTEST_LOGGING`: Log level used to report in pytest.log
+
+## Tests Results
+
+Results are printed to your screen and also saved in reports:
+
+- `report.html`: Pytest result with logging
+- `htmlcov/index.html`: Coverage report
+
+> More information on pytest, can be found [here](https://docs.pytest.org/)
```

Comment:
Similar to above, update sentence structure on links. See https://developers.google.com/style/link-text